source/MaterialXGenMsl/MslShaderGenerator.cpp

```diff
@@ -0,0 +1,1430 @@
+                {
+                    emitLine(vertex_inputs.getName() + " " + vertex_inputs.getInstance(), stage, !globalContextConstructorParams);
+                    if(globalContextConstructorParams)
+                        separator = ", ";
+                    else
+                        separator = "\n";
+                }
+                else if (globalContextConstructorInit)
+                {
+                    emitLine(vertex_inputs.getInstance() + "(" + vertex_inputs.getInstance() + ")", stage, !globalContextConstructorInit);
+                    separator = ", ";
+                }
+            }
+        }
+        else
+        {
+            emitString(vertex_inputs.getName() + " " + vertex_inputs.getInstance() + " [[ stage_in ]]", stage);
+            separator = ", ";
+        }
+    }
+    
+    // Add all uniforms
+    for (const auto& it : stage.getUniformBlocks())
+    {
+        const VariableBlock& uniforms = *it.second;
+        bool isLightData = uniforms.getName() == HW::LIGHT_DATA;
+        
+        if(!needsLightData && isLightData) continue;
+        
+        if(isLightData)
+        {
+            emitString(separator, stage);
+            if(entryFunctionArgs)
+            {
+                emitString(_syntax->getUniformQualifier() + " " +
+                           uniforms.getName() + "_" + stage.getName() + "& " +
+                           uniforms.getInstance() +
+                           "[[ buffer(" + std::to_string(buffer_slot++) + ") ]]",
+                           stage);
+            }
+            else if(globalContextInit)
+            {
+                emitLine(uniforms.getInstance() + "." + uniforms.getInstance(), stage, false);
+            }
+            else if(globalContextMembers || globalContextConstructorParams)
+            {
+                const string structArraySuffix = "[" + HW::LIGHT_DATA_MAX_LIGHT_SOURCES + "]";
+                emitLine((globalContextConstructorParams ? (_syntax->getUniformQualifier() + " ") : std::string()) +
+                         uniforms.getName() + " " +
+                         uniforms.getInstance() +
+                         structArraySuffix, stage, !globalContextConstructorParams);
+            }
+            else if(globalContextConstructorInit)
+            {
+                const unsigned int maxLights = std::max(1u, context.getOptions().hwMaxActiveLightSources);
+                emitLine(uniforms.getInstance(), stage, false);
+                emitScopeBegin(stage);
+                for(unsigned int l = 0; l < maxLights; ++l)
+                {
+                    emitString(l == 0 ? "" : ", ", stage);
+                    emitLine(uniforms.getInstance() +
+                             "["+ std::to_string(l)  +"]",
+                             stage, false);
+                }
+                emitScopeEnd(stage);
+            }
+        }
+        else
+        
+        if(!entryFunctionArgs)
```

Comment:
Was this intended as an `else if` statement, or is there a missing `else` clause that should be added here?  If it's intended as `else if`, then I'd recommend merging it into a single line for clarity.

Reply:
In fact all if-else statement was supposed to be in else. I added curly braces to make else boundary more clear. Thanks for the comment 👍